.azure-pipelines.yml

```diff
@@ -26,6 +25,71 @@ jobs:
       done
     name: test_jobs
     displayName: 'Run ./wpt test-jobs'
+  - script: |
+      # This assumes that refs/pull/1234/merge has been checked out, and that
+      # the first parent is the target branch (usually master) and the second
+      # parent the PR branch.
+      echo Build.SourceBranch=$(Build.SourceBranch)
+      echo Build.SourceBranchName=$(Build.SourceBranchName)
+      echo Build.SourceVersion=$(Build.SourceVersion)
+      echo System.PullRequest.SourceBranch=$(System.PullRequest.SourceBranch)
+      echo System.PullRequest.TargetBranch=$(System.PullRequest.TargetBranch)
+      MERGE_COMMIT=`git rev-parse HEAD`
+      echo MERGE_COMMIT=$MERGE_COMMIT
+      FIRST_PARENT=`git rev-parse HEAD^1`
+      echo FIRST_PARENT=$FIRST_PARENT
+      SECOND_PARENT=`git rev-parse HEAD^2`
+      echo SECOND_PARENT=$SECOND_PARENT
+      echo "##vso[task.setvariable variable=Merge;isOutput=true]$MERGE_COMMIT";
+      echo "##vso[task.setvariable variable=TargetBranch;isOutput=true]$FIRST_PARENT";
+      echo "##vso[task.setvariable variable=SourceBranch;isOutput=true]$SECOND_PARENT";
+    name: commits
+    displayName: 'Record commits'
+
+- job: affected_macOS
+  displayName: 'affected tests (Safari Technology Preview)'
+  #dependsOn: root
+  #condition: dependencies.root.outputs['test_jobs.stability']
+  pool:
+    vmImage: 'macOS-10.13'
+  steps:
+  - template: tools/ci/azure/checkout.yml
+  - template: tools/ci/azure/pip_install.yml
+    parameters:
+      packages: virtualenv
+  - template: tools/ci/azure/install_fonts.yml
+  - template: tools/ci/azure/install_certs.yml
+  - template: tools/ci/azure/install_safari.yml
+  - template: tools/ci/azure/update_hosts.yml
+  - script: no_proxy='*' ./wpt run --yes --no-pause --no-fail-on-unexpected --no-restart-on-unexpected --log-wptreport $(Build.ArtifactStagingDirectory)/wpt_report.json --channel preview safari --affected HEAD^1
+    displayName: 'Run tests'
+  - task: PublishBuildArtifacts@1
```

Comment:
According to the [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/publish-build-artifacts?view=vsts), we can explicitly specify `pathtoPublish` to a *file* instead of the default directory. Can we try that?

Reply:
Sure, I'll try and see what happens